src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCountProperly.cs

```diff
@@ -0,0 +1,620 @@
+                            shouldNegate = false;
+                            break;
+
+                        default:
+                            return false;
+                    }
+                    break;
+                default:
+                    return false;
+            }
+
+            return true;
+        }
+
+        private static bool TypeContainsVsibileProperty(OperationAnalysisContext context, ITypeSymbol type, string propertyName, SpecialType propertyType)
+            => TypeContainsVsibileProperty(context, type, propertyName, propertyType, propertyType);
+
+        private static bool TypeContainsVsibileProperty(OperationAnalysisContext context, ITypeSymbol type, string propertyName, SpecialType lowerBound, SpecialType upperBound)
+        {
+            if (type.GetMembers(propertyName).FirstOrDefault() is IPropertySymbol property)
+            {
+                return IsPropertyValidAndVisible(context, property, lowerBound, upperBound);
+            }
+
+            // The property might not be defined on the specified type if the type is an interface, it can be defined in one of the parent interfaces.
+            if (type.TypeKind == TypeKind.Interface)
+            {
+                foreach (var @interface in type.AllInterfaces)
+                {
+                    if (@interface.GetMembers(propertyName).FirstOrDefault() is IPropertySymbol interfaceProperty)
+                    {
+                        return IsPropertyValidAndVisible(context, interfaceProperty, lowerBound, upperBound);
+                    }
+                }
+            }
+            else
+            {
+                ITypeSymbol? currentType = type.BaseType;
+                while (currentType != null)
+                {
+                    if (currentType.GetMembers(propertyName).FirstOrDefault() is IPropertySymbol typeProperty)
+                    {
+                        return IsPropertyValidAndVisible(context, typeProperty, lowerBound, upperBound);
+                    }
+
+                    currentType = currentType.BaseType;
+                }
+            }
+
+            return false;
+        }
+
+        private static bool IsPropertyValidAndVisible(OperationAnalysisContext context, IPropertySymbol property, SpecialType lowerBound, SpecialType upperBound)
+            => !property.IsStatic &&
+            IsInRangeInclusive((uint)property.Type.SpecialType, (uint)lowerBound, (uint)upperBound) &&
+            property.GetMethod != null &&
+            context.Compilation.IsSymbolAccessibleWithin(property, context.ContainingSymbol.ContainingType) &&
+            context.Compilation.IsSymbolAccessibleWithin(property.GetMethod, context.ContainingSymbol.ContainingType);
+
+        private static bool TryGetZeroOrOneConstant(IOperation operation, out int constant)
+        {
+            constant = default;
+
+            if (operation?.Type?.SpecialType != SpecialType.System_Int32 &&
```

Comment:
Nit: Can we use a switch statement/expression here and simplify code?

Reply:
Seems like this is moving existing code. Feel free to retain or change as appropriate.